x/operator/keeper/abci.go

```diff
@@ -90,10 +107,90 @@ func (k *Keeper) UpdateVotingPower(ctx sdk.Context, avsAddr string) error {
 	if err != nil {
 		return err
 	}
+
+	// TODO: Consider not addressing the dogfood AVS, as its historical voting power
+	// has already been stored by CometBFT.
+
+	// set voting power snapshot
+	// When the snapshot helper does not exist, it represents the initial state of AVS,
+	// where no snapshot information has been stored. Therefore, it is necessary to store
+	// both the snapshot and the helper information.
+	var snapshotHelper operatortypes.SnapshotHelper
+	if !k.HasSnapshotHelper(cc, avsAddr) {
+		isSnapshotChanged = true
+	} else {
+		snapshotHelper, err = k.GetSnapshotHelper(cc, avsAddr)
+		if err != nil {
+			return err
+		}
+	}
+	votingPowerSnapshot := operatortypes.VotingPowerSnapshot{
+		EpochIdentifier: epochIdentifier,
+		EpochNumber:     epochNumber,
+		BlockTime:       ctx.BlockTime(),
+	}
+
+	// The voting power calculated at the end of the current epoch will be applied
+	// to the next epoch. Therefore, when storing the voting power snapshot, we use
+	// the `start_height` of the next epoch as the key. This ensures that during the
+	// slashing process, there is no need to account for voting power activation delay;
+	// it can be used directly.
+	// Use the current height as the snapshot height when handling snapshots triggered
+	// by slashing. This prevents stakers from escaping slashes through backrunning
+	// undelegation.
+	snapshotHeight := ctx.BlockHeight()
+	if !isForSlash {
+		// clear the slash flag at the end of the epoch
+		snapshotHelper.HasSlash = false
+		// Use the start height of the next epoch as the snapshot key.
+		// The start height of the next epoch should be the current height plus 1,
+		// as voting power is updated at the end of the epoch.
+		snapshotHeight++
+		votingPowerSnapshot.EpochNumber++
+	}
+	if snapshotHelper.HasOptOut || isSnapshotChanged {
+		votingPowerSnapshot.TotalVotingPower = avsVotingPower
+		votingPowerSnapshot.VotingPowerSet = votingPowerSet
+		snapshotHelper.LastChangedHeight = snapshotHeight
+		// clear the hasOptOut flag if it's certain that the snapshot will be updated
+		snapshotHelper.HasOptOut = false
+	}
+	votingPowerSnapshot.LastChangedHeight = snapshotHelper.LastChangedHeight
+
+	err = k.SetSnapshotHelper(cc, avsAddr, &snapshotHelper)
+	if err != nil {
+		return err
+	}
+	snapshotKey := assetstype.GetJoinedStoreKey(avsAddr, strconv.FormatInt(snapshotHeight, 10))
```

Comment:
This is not sortable in the iterator. Please use the bytes format as `sdk.Uint64ToBigEndian(height)`

Reply:
Yes, it has been fixed in the latest commit.